src/components/Contacts/ContactDetails/ContactDetailsHeader/DeleteContactModal/DeleteContactModal.tsx

```diff
@@ -12,37 +17,210 @@ import {
   DeleteButton,
 } from 'src/components/common/Modal/ActionButtons/ActionButtons';
 import Modal from 'src/components/common/Modal/Modal';
+import { useAccountListId } from 'src/hooks/useAccountListId';
+import { isEditableSource, sourceToStr } from 'src/utils/sourceHelper';
+import { useContactSourceQuery } from './ContactSource.generated';
 
 const LoadingIndicator = styled(CircularProgress)(({ theme }) => ({
   margin: theme.spacing(0, 1, 0, 0),
 }));
 
+interface DataInfo {
+  canDelete: boolean;
+  uniqueSources: {
+    contact: string[];
+    address: string[];
+    email: string[];
+    phone: string[];
+  };
+}
 interface DeleteContactModalProps {
   open: boolean;
   setOpen: (open: boolean) => void;
   deleting: boolean;
   deleteContact: () => void;
+  contactId: string;
 }
 
 export const DeleteContactModal: React.FC<DeleteContactModalProps> = ({
   open,
   setOpen,
   deleting,
   deleteContact,
+  contactId,
 }) => {
   const { t } = useTranslation();
+  const accountListId = useAccountListId() ?? '';
+
+  const { data } = useContactSourceQuery({
+    variables: { accountListId, contactId },
+    skip: !open && !contactId,
+  });
+  const contactSources = data?.contact;
+
+  const dataInfo: DataInfo = useMemo(() => {
+    if (!contactSources) {
+      return {
+        canDelete: true,
+        uniqueSources: {
+          contact: [],
+          address: [],
+          email: [],
+          phone: [],
+        },
+      };
+    }
+    // We ensure the contact was created on MPDX and that all the data is editable.
+    // If any data is not editable, this means it was created by a third party.
+    // Which will only recreate the data after deleting it on MPDX.
+    // To prevent this confusion, we do not allow a contact to be deleted if it has non editable data.
+
+    const isContactNonEditable = !isEditableSource(contactSources.source ?? '');
+
+    const sources: Map<string, string[]> = new Map();
+    sources.set('contact', [contactSources.source ?? '']);
+
+    const isAddressNonEditable = contactSources.addresses?.nodes.reduce(
+      (acc, address) => {
+        sources.set('address', [
+          ...(sources.get('address') ?? []),
+          address.source,
+        ]);
+        return acc || !isEditableSource(address.source ?? '');
```

Comment:
Address sources aren't nullable

```suggestion
        return acc || !isEditableSource(address.source);
```

Reply:
Looks like I missed that one.